quickwit/quickwit-janitor/src/retention_policy_execution.rs

```diff
@@ -70,145 +80,36 @@ pub async fn run_execute_retention_policy(
     Ok(expired_splits)
 }
 
-/// Checks to see if a split is expired based on a retention policy.
-fn is_split_expired(
-    current_date_time: OffsetDateTime,
-    split: &Split,
-    retention_policy: &RetentionPolicy,
-) -> anyhow::Result<bool> {
-    let retention_period = retention_policy.retention_period()?;
-    // The split `publish_timestamp` field does not exist in previous versions of Quickwit (< v0.4).
-    // In order to stay backward compatible, we will use the `updated_timestamp` field
-    // for splits generated from older version of Quickwit.
-    let publish_timestamp = split.publish_timestamp.unwrap_or(split.update_timestamp);
-    let publish_date_time = OffsetDateTime::from_unix_timestamp(publish_timestamp)?;
-    let is_expired = match retention_policy.cutoff_reference {
-        RetentionPolicyCutoffReference::PublishTimestamp => {
-            (current_date_time - publish_date_time) >= retention_period
-        }
-        RetentionPolicyCutoffReference::SplitTimestampField => {
-            if let Some(time_range) = &split.split_metadata.time_range {
-                let time_range_end_date_time =
-                    OffsetDateTime::from_unix_timestamp(*time_range.end())?;
-                (current_date_time - time_range_end_date_time) >= retention_period
-            } else {
-                warn!(index_id=%split.split_metadata.index_id, split_id=%split.split_id(),
-                "Retention policy execution expected a `time_range` on the split, but none exist.");
-                false
-            }
-        }
-    };
-    Ok(is_expired)
-}
-
-#[cfg(test)]
-mod tests {
```

Comment:
No need for a test?

Reply:
The tests here were doing what we now do in the metastore. I added a test on metastore to compensate.